network/p2p/libp2pNode_test.go

```diff
@@ -86,297 +39,148 @@ func (suite *LibP2PNodeTestSuite) TestMultiAddress() {
-		SetConnectionManager(connManager).
-		SetPingInfoProvider(pingInfoProvider).
-		SetResolver(resolver).
-		SetTopicValidation(false).
-		SetStreamCompressor(WithGzipCompression).
-		SetLogger(log)
-
-	if allowList {
-		connGater := NewConnGater(log)
-		builder.SetConnectionGater(connGater)
-	}
-
-	ctx := context.Background()
-	n, err := builder.Build(ctx)
-	require.NoError(t, err)
-
-	n.SetFlowProtocolStreamHandler(handlerFunc)
-
-	require.Eventuallyf(t, func() bool {
-		ip, p, err := n.GetIPPort()
-		return err == nil && ip != "" && p != ""
-	}, 3*time.Second, tickForAssertEventually, fmt.Sprintf("could not start node %s", identity.NodeID.String()))
-
-	// get the actual IP and port that have been assigned by the subsystem
-	ip, port, err := n.GetIPPort()
-	require.NoError(t, err)
-	identity.Address = ip + ":" + port
-
-	return n, *identity
-}
-
-func MockPingInfoProvider() (*mocknetwork.PingInfoProvider, string, uint64, uint64) {
-	version := "version_1"
-	height := uint64(5000)
-	view := uint64(10)
-	pingInfoProvider := new(mocknetwork.PingInfoProvider)
-	pingInfoProvider.On("SoftwareVersion").Return(version)
-	pingInfoProvider.On("SealedBlockHeight").Return(height)
-	pingInfoProvider.On("HotstuffView").Return(view)
-	return pingInfoProvider, version, height, view
-}
-
-// StopNodes stop all nodes in the input slice
-func StopNodes(t *testing.T, nodes []*Node) {
-	for _, n := range nodes {
-		StopNode(t, n)
-	}
-	fmt.Println("[debug] nodes stopped")
-}
-
-func StopNode(t *testing.T, node *Node) {
-	addr := node.host.Addrs()
-	fmt.Printf("[debug] stopping %v\n", addr)
-	done, err := node.Stop()
-	assert.NoError(t, err)
-	<-done
-	fmt.Printf("[debug] stopped %v\n", addr)
-}
-
-// generateNetworkingKey is a test helper that generates a ECDSA flow key pair.
-func generateNetworkingKey(t *testing.T) fcrypto.PrivateKey {
-	seed := unittest.SeedFixture(48)
-	key, err := fcrypto.GeneratePrivateKey(fcrypto.ECDSASecp256k1, seed)
-	require.NoError(t, err)
-	return key
-}
-
-// silentNodeFixture returns a TCP listener and a node which never replies
-func silentNodeFixture(t *testing.T) (net.Listener, flow.Identity) {
-	key := generateNetworkingKey(t)
-
-	lst, err := net.Listen("tcp4", ":0")
-	require.NoError(t, err)
-
-	addr, err := manet.FromNetAddr(lst.Addr())
-	require.NoError(t, err)
-
-	addrs := []multiaddr.Multiaddr{addr}
-	addrs, err = addrutil.ResolveUnspecifiedAddresses(addrs, nil)
-	require.NoError(t, err)
-
-	go acceptAndHang(t, lst)
-
-	ip, port, err := IPPortFromMultiAddress(addrs...)
-	require.NoError(t, err)
-
-	identity := unittest.IdentityFixture(unittest.WithNetworkingKey(key.PublicKey()), unittest.WithAddress(ip+":"+port))
-	return lst, *identity
-}
-
-func acceptAndHang(t *testing.T, l net.Listener) {
-	conns := make([]net.Conn, 0, 10)
-	for {
-		c, err := l.Accept()
-		if err != nil {
-			break
-		}
-		if c != nil {
-			conns = append(conns, c)
-		}
-	}
-	for _, c := range conns {
-		require.NoError(t, c.Close())
-	}
-}
```

Comment:
moved to `fixtures.go` as shared between several test files. Also, did some refactoring to extend these shared fixtures among more tests, hence removing their redundancies.